index.js

```diff
@@ -72,11 +72,15 @@ module.exports = {
 
   sassOptions: function () {
     var env  = process.env.EMBER_ENV;
-    var options = (this.app && this.app.options && this.app.options.sassOptions) || {};
-    var parentOption = (this.parent && this.parent.app && this.parent.app.options && this.parent.app.options.sassOptions) || {};
     var envConfig = this.project.config(env).sassOptions;
 
-    Object.assign(options, parentOption);
+    // *Either* use the options for an addon which is consuming this, *or* for
+    // an app which is consuming this, but never *both* at the same time. The
+    // special case here is `"dummy"`, which only applies when testing an addon.
+    var useApp = this.app.name === 'dummy' || this.app.name === this.project.name();
```

Comment:
I think I would use `this.isDevelopingAddon()` here instead of the manual checks for `dummy`...

Reply:
Ah, perfect.